contract/src/main/kotlin/com/ritense/valtimo/contract/dashboard/QueryCondition.kt

```diff
@@ -0,0 +1,147 @@
+            it.isNotEmpty() && it.startsWith("\${") && it.endsWith('}') && "localDateTimeNow" in it
+        } ?: false
+    }
+
+    private fun getPredicateFromDateTimeSpelExpression(
+        root: Root<*>,
+        it: QueryCondition<String>,
+        criteriaBuilder: CriteriaBuilder,
+        pathExpressionFunction: (Class<T>, String, Root<*>, CriteriaBuilder) -> Expression<T>
+    ): Predicate {
+        val parser = SpelExpressionParser()
+        val expressionWithoutPrefixSuffix = it.queryValue.substringAfter("\${").substringBefore("}")
+
+        val spelEvaluationContext = WidgetDataSourceSpelEvaluationContext()
+        val context = StandardEvaluationContext()
+
+        context.setRootObject(spelEvaluationContext)
+
+        val spelExpression: org.springframework.expression.Expression =
+            parser.parseExpression(expressionWithoutPrefixSuffix)
+
+        val valueClass = LocalDateTime::class.java
+
+        val value = spelExpression.getValue(context, valueClass)
+
+        val expression =
+            pathExpressionFunction(valueClass as Class<T>, it.queryPath, root, criteriaBuilder)
+
+        return it.queryOperator.toPredicate(
+            criteriaBuilder,
+            expression,
+            value as T
+        )
+    }
+
+    private fun <T> queryValueIsCurrentUserExpression(target: T): Boolean {
+        return (target as? String)?.let {
+            it.isNotEmpty() &&
+                PermissionConditionKey.isValidKey(it) &&
+                // roles is a list and is currently not supported
+                PermissionConditionKey.fromKey(it) != PermissionConditionKey.CURRENT_USER_ROLES
+        } ?: false
+    }
+
+    private fun getPredicateFromCurrentUserExpression(
+        root: Root<*>,
+        it: QueryCondition<String>,
+        criteriaBuilder: CriteriaBuilder,
+        pathExpressionFunction: (Class<T>, String, Root<*>, CriteriaBuilder) -> Expression<T>
+    ): Predicate {
+        val valueClass = String::class.java
+        val expression = pathExpressionFunction(valueClass as Class<T>, it.queryPath, root, criteriaBuilder)
+        val permissionConditionKey = PermissionConditionKey.fromKey(it.queryValue)?.key
+        val resolvedValue = CurrentUserExpressionHandler.resolveValue(permissionConditionKey) as? String ?: ""
+
+        return it.queryOperator.toPredicate(
+            criteriaBuilder,
+            expression,
+            resolvedValue as T
+        )
+    }
+
+
+    fun toPredicate(
+        root: Root<*>,
+        criteriaBuilder: CriteriaBuilder,
+        pathExpressionFunction: (Class<Any>, String, Root<*>, CriteriaBuilder) -> Expression<Any>
+    ): Predicate {
+        pathExpressionFunction as (Class<T>, String, Root<*>, CriteriaBuilder) -> Expression<T>
+
+        if (queryValueIsDateTimeSpelExpression(queryValue)) {
+            return getPredicateFromDateTimeSpelExpression(
+                root,
+                this as QueryCondition<String>,
```

Comment:
Why do we need to pass this in if it is always already the current object and the method is in the object?